internal/common/physics/engine.go

```diff
@@ -56,6 +59,7 @@ func GenerateNewState(initialState utils.PhysicalState, force float64, orientati
 		Position:     coordinates,
 		Acceleration: acceleration,
 		Velocity:     velocity,
+		Mass:         initialState.Mass,
```

Comment:
I am actually quite upset that this went unnoticed. There is no way anyone tested this code